### DISCUSSION TITLE
2 NOOB Questions - arcOS

### DISCUSSION THREAD

**Discussion by N1JCM** [kg4vdk/arcos-linux-modules#14] [General Discussion]
Hello All - New user questions (after seeing the KM4ACK interview and demo of arcOS).

1. I have created the arcOS 21.3-QRV Foxtrot USB boot on a 16K USB drive. It boots and runs on a Core2 Duo laptop. Upon boot, the startup screen says a persistent storage device has been created, reboot to implement. Lower right says no persistent storage. After a restart from the power menu item, arcOS reboots, but the same persistent storage message is shown and lower right still says no persistent storage. 4 restarts so far without persistent storage being created. I suspect user error, but hope for suggestions from the group.

2. Any way to get FLRIG into the app list? I have run FLDIGI and did the Hamlib setup successfully, but wanted FLRIG so I can have a single rig control for FLDIGI, JS8CALL (and maybe WSJT-X).

Comments and questions welcome & thanks!

--
Jeff Marden/N1JCM

**Comment by kg4vdk** [comment]
Jeff, 

For the first part can you provide a few more details please?

1. Confirm that the MD5SUM for the downloaded ISO matches [this one](https://s3.amazonaws.com/arcos-linux.com/arcOS-21.3_QRV_Foxtrot.md5.txt).
2. How did you write the ISO to the USB? Which application, and any unique options?
3. Is this a "known good" USB, or possibly an old spare "unknown"?

For further insight, when the system boots but you receive the unexpected Persistent Storage dialog...

1. Choose "Cancel"
2. Open a terminal (CTL+ALT+T), and run `persistent-storage`.
3. Please copy any output here (using the code tags).

For the second part, about FLRIG. FLRIG is included, and can be found in the "Amateur Radio" section of the Main Menu.

Thanks for trying it out! We can turn the 1st part back into an Issue if necessary, once we've got a better handle on the particulars.

73 de KG4VDK
Mike

**Comment by N1JCM** [comment]
Hello Mike: Thx for follow-up.
Details - 1. I'll do the MD5sum tomorrow, but the arcOS boots and runs several applications OK, so I suspect the download was OK.
               2. I used RUFUS to write the ISO to the USB drive. It did ask for two updated linux files for before writing, but no other special commands and no errors. Other, more appropriate ISO write utility?
               3.  New 16GB USB drive, still in the package. Not "known good" but new.
             
Will-do on the terminal - persistent-storage utility.

Lastly, thanks for the pointer. I did not look past the buttons along the bottom of the screen.
--
Jeff Marden/N1JCM

**Reply by kg4vdk** [reply]
No worries. I also doubt the download was corrupted based on the booting/partially working. As for Rufus, I've never tested it. But, I have heard through the grapevine on the Mint forums that Rufus sometimes has bad outcomes. I have tested with the [Raspberry Pi Imager](https://www.raspberrypi.com/software/). You can use the "Custom Image" option at the bottom of the scrolling dropdown for selecting the OS to write.

The "two updated Linux files before writing" have me thinking that's likely where the issue lies. Keep me updated!

**Comment by N1JCM** [comment]
Hello Mike: For the Persistent Storage issue...

I hit cancel from the notice screen.

Cntl-Shift-T - No action.
Instead I ran the terminal from the Menu. 
Type "persistent-storage" @ prompt =>

fdisk: bad usage
Try 'fdisk --help' for more information
exfatprogs version : 1.1.3
open failed : 3, No such file or directory

exFAT format fail!

I'm starting to suspect that Rufus did not use exFAT file format in creating the arcOS USB Drive.

Thoughts?

-- Jeff Marden/N1JCM

**Reply by kg4vdk** [reply]
Thanks for that, Jeff!

TL;DR - Use the "DD" method in Rufus, with no other modifications to the recommended defaults.

I did some testing on my end with Rufus (v4.5). When I used the "click through using all recommended defaults" method, I was able to replicate the issue you experienced. When I used the "DD" method in Rufus, everything went smoothly.
![Rufus ISO or DD](https://github.com/user-attachments/assets/f6cdc77c-f6f6-4a44-b7a9-12aa57e43fed)


Rufus essentially tries to do some magic that is handled in the ISO image already. `isohybrid` images already contain the syslinux files Rufus inserts using the recommended defaults.

When the `persistent-storage` runs at boot, it checks to see if there is a third partition on the boot device, as a "dd" type install would have created (One partition is the ISO image, and the second is an EFI partition to bridge the gap for booting on both legacy BIOS and EFI). If that third, unused, partition exists, it then formats that partition as ARCOS-DATA. It fails after the Rufus install because Rufus only leaves two partitions on the disk, so the attempt to format /dev/xxx3 as exFAT fails. This should have an error message to more clearly let the user know what has happened, and possibly suggest a cause/resolution. Adding that functionality would require the ISO to be rebuilt, so I will add it to the next release. I don't think it's a showstopper for Foxtrot.

**Comment by N1JCM** [comment]
Hello Mike - Thanks for the extra digging into RUFUS. Using the DD selection worked and upon boot and reboot the ARCOS-DATA partition was created and all appears OK. I ran FLDIGI and connected to my IC-7100 fine; config items saved etc.
The USB boot is fine but I have a spare laptop and would like to consider loading the ISO on the HD. Any hints on how to do this?
Thanks again - Jeff Marden/N1JCM
